pybamm/expression_tree/functions.py

```diff
@@ -87,11 +89,11 @@ def diff(self, variable):
 
             derivative = sum(partial_derivatives)
             if derivative == 0:
-                derivative = pybamm.Scalar(0)
+                return pybamm.Scalar(0)
 
             return derivative
 
-    def _function_diff(self, children, idx):
+    def _function_diff(self, children: Sequence[pybamm.Symbol], idx: float):
```

Comment:
In a previous comment I mentioned `Iterable`, perhaps this should be `Sequence`. idk the difference..?

Reply:
Any `Sequence` is an `Iterable` - but `Sequence` objects must define `_len_`. As `len(children)` is called in this function, Sequence is appropriate.